streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/service/EnvironmentService.java

```diff
@@ -434,6 +439,85 @@ public NamespaceServiceClusterMapping addOrUpdateServiceClusterMapping(
         return newMapping;
     }
 
+    public void injectProtocolAndPortToComponent(Map<String, String> configurations, Component component) {
+        try {
+            ComponentPropertyPattern confMap = ComponentPropertyPattern
+                    .valueOf(component.getName());
+            String value = configurations.get(confMap.getConnectionConfName());
+            if (value != null) {
+                Matcher matcher = confMap.getParsePattern().matcher(value);
+
+                if (matcher.matches()) {
+                    String protocol = matcher.group(1);
+                    String portStr = matcher.group(2);
+
+                    if (!protocol.isEmpty()) {
+                        component.setProtocol(protocol);
+                    }
+                    if (!portStr.isEmpty()) {
+                        try {
+                            component.setPort(Integer.parseInt(portStr));
+                        } catch (NumberFormatException e) {
+                            LOG.warn(
+                                    "Protocol/Port information [{}] for component {} doesn't seem to known format [{}]."
+                                            + "skip assigning...", value, component.getName(), confMap.getParsePattern());
+
+                            // reset protocol information
+                            component.setProtocol(null);
+                        }
+                    }
+                } else {
+                    LOG.warn("Protocol/Port information [{}] for component {} doesn't seem to known format [{}]. "
+                            + "skip assigning...", value, component.getName(), confMap.getParsePattern());
+                }
+            } else {
+                LOG.warn("Protocol/Port related configuration ({}) is not set", confMap.getConnectionConfName());
+            }
+        } catch (IllegalArgumentException e) {
+            // don't know port related configuration
+        }
+    }
+
+    public Collection<ServiceBundle> listServiceBundles(List<QueryParam> params) {
+        return dao.find(SERVICE_BUNDLE_NAMESPACE, params);
+    }
+
+    public ServiceBundle getServiceBundleByName(String serviceName) {
+        Collection<ServiceBundle> serviceBundles = listServiceBundles(Lists.newArrayList(new QueryParam("name", serviceName)));
+        if (serviceBundles.size() > 1) {
+            LOG.warn("Multiple service bundles have same name: {} returning first match.", serviceName);
+            return serviceBundles.iterator().next();
+        } else if (serviceBundles.size() == 1) {
+            return serviceBundles.iterator().next();
+        }
+        return null;
```

Comment:
It should be consistent between getXXX with DAO, and I didn't see any of them throwing exception.